types/index.d.ts

```diff
@@ -0,0 +1,94 @@
+import Vue, { VNodeData, Component, ComponentOptions, FunctionalComponentOptions } from 'vue'
+
+/**
+ * Utility type to declare an extended Vue constructor
+ */
+type VueClass<V extends Vue> = (new (...args: any[]) => V) & typeof Vue
+
+/**
+ * Utility type for a selector
+ */
+type Selector = string | Component
+
+/**
+ * Utility type for slots
+ */
+type Slots = {
+  [key: string]: (Component | string)[] | Component | string
+}
+
+/**
+ * Utility type for stubs which can be a string of template as a shorthand
+ * If it is an array of string, the specified children are replaced by blank components
+ */
+type Stubs = {
+  [key: string]: Component | string | boolean
+} | string[]
+
+/**
+ * Base class of Wrapper and WrapperArray
+ * It has common methods on both Wrapper and WrapperArray
+ */
+interface BaseWrapper {
+  contains (selector: Selector): boolean
+  exists (): boolean
+
+  hasAttribute (attribute: string, value: string): boolean
+  hasClass (className: string): boolean
+  hasProp (prop: string, value: any): boolean
+  hasStyle (style: string, value: string): boolean
+
+  is (selector: Selector): boolean
+  isEmpty (): boolean
+  isVueInstance (): boolean
+
+  update (): void
+  setData (data: object): void
+  setProps (props: object): void
+  trigger (eventName: string, options?: object): void
+}
+
+interface Wrapper<V extends Vue> extends BaseWrapper {
+  readonly vm: V
+  readonly element: HTMLElement
+  readonly options: WrapperOptions
+
+  find<R extends Vue> (selector: Selector): Wrapper<R>
+  findAll (selector: Selector): WrapperArray
+
+  html (): string
+  text (): string
+  name (): string
+}
+
+interface WrapperArray extends BaseWrapper {
+  readonly length: number
+
+  at<V extends Vue> (index: number): Wrapper<V>
+}
+
+interface WrapperOptions {
+  attachedToDocument: boolean
+}
+
+interface MountOptions<V extends Vue> extends ComponentOptions<V> {
```

Comment:
I think vuejs/vue#5887 will not be merged soon since we need to have some migration time for the new declaration. So we can use the current declaration for now.